### PR TITLE
Feat/homepage

### PR DESCRIPTION
**Pull request by Megavex**

### CODE REVIEW

File: application/app/src/main/java/com/yahyeet/boardbook/activity/HomeActivity.java [hunk 1]
```diff
@@ -1,64 +1,53 @@
...
-import android.os.Bundle;
-import android.view.View;
-
 import com.yahyeet.boardbook.R;
 
 public class HomeActivity extends AppCompatActivity {
 
-    
+
+
```
Comment (cajoho99) [inline application/app/src/main/java/com/yahyeet/boardbook/activity/HomeActivity.java]: Too many newlines

File: application/app/src/main/java/com/yahyeet/boardbook/activity/HomeActivity.java [hunk 2]
```diff
@@ -1,64 +1,53 @@
...
-                    break;
-                case R.id.nav_chat:
-                    selectedFragment = new ChatFragment();
-                    break;
-            }
-            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,selectedFragment).commit();
-            return true;
-        }
-    };
+
```
Comment (cajoho99) [inline application/app/src/main/java/com/yahyeet/boardbook/activity/HomeActivity.java]: Too many newlines

File: application/app/src/main/java/com/yahyeet/boardbook/activity/HomeFragment.java [hunk 3]
```diff
@@ -8,14 +9,52 @@
...
 
 import com.yahyeet.boardbook.R;
+import com.yahyeet.boardbook.presenter.HomePresenter;
 
-public class HomeFragment extends Fragment {
+import javax.annotation.Nonnull;
+
+public class HomeFragment extends Fragment implements IHomeFragment{
+
+
```
Comment (cajoho99) [inline application/app/src/main/java/com/yahyeet/boardbook/activity/HomeFragment.java]: Too many newlines

File: application/app/src/main/java/com/yahyeet/boardbook/activity/HomeFragment.java [hunk 4]
```diff
@@ -8,14 +9,52 @@
...
+    /**
+     * Orders recyclerView of matches to repopulate itself
+     */
+    public void repopulateMatchFeed(){
+        homePresenter.repopulateMatches();
+    }
+
+
+
+
```
Comment (cajoho99) [inline application/app/src/main/java/com/yahyeet/boardbook/activity/HomeFragment.java]: newlines?

File: application/app/src/main/java/com/yahyeet/boardbook/presenter/adapter/MatchAdapter.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        private TextView dateText;
+        private ImageView imageView;
+
+
+        MatchViewHolder(View v) {
+            super(v);
+            // Define click listener for the ViewHolder's View.
+            v.setOnClickListener(v1 -> Log.d(TAG, "Element " + getAdapterPosition() + " clicked."));
+
+
```
Comment (cajoho99) [inline application/app/src/main/java/com/yahyeet/boardbook/presenter/adapter/MatchAdapter.java]: remove newline

File: application/app/src/main/java/com/yahyeet/boardbook/activity/HomeActivity.java [hunk 6]
```diff
@@ -1,83 +1,62 @@
...
-
-
 
     /**
      * This method when called starts a new login activity
+     *
      * @param view is the visual object (ex a button) the method is bound to (I think)
      */
-    public void ChangeToLogin(View view){
+    public void ChangeToLogin(View view) {
```
Comment (rosengrenen) [inline application/app/src/main/java/com/yahyeet/boardbook/activity/HomeActivity.java]: camelCase pl0x

File: application/app/src/main/java/com/yahyeet/boardbook/activity/IHomeFragment.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.yahyeet.boardbook.activity;
+
+public interface IHomeFragment {
+
```
Comment (rosengrenen) [inline application/app/src/main/java/com/yahyeet/boardbook/activity/IHomeFragment.java]: remove lines pl0x

File: application/app/src/main/java/com/yahyeet/boardbook/activity/HomeActivity.java (hunk 6 above)
Comment (cajoho99) [reply application/app/src/main/java/com/yahyeet/boardbook/activity/HomeActivity.java]: Fixed

File: application/app/src/main/java/com/yahyeet/boardbook/activity/IHomeFragment.java (hunk 7 above)
Comment (cajoho99) [reply application/app/src/main/java/com/yahyeet/boardbook/activity/IHomeFragment.java]: Fixed

**Pull request merged**